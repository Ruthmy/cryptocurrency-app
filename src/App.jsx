import React from 'react';
import { HashRouter, Routes, Route } from 'react-router-dom';

import Navbar from './components/Navbar';

import './App.css';

const App = () => (
  <>
    <HashRouter>
      <Navbar />

      <Routes>
        <Route path="/" element={<div>Home Page</div>} />
        <Route path="/calculator" element={<div>Details Page</div>} />
        <Route path="*" element={<p>Not found</p>} />
      </Routes>
    </HashRouter>
  </>
);

export default App;
