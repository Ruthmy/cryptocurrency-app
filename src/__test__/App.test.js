import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom';

import App from '../App';

const mockStore = configureStore([]);

describe('App', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      weather: {
        weather: [{
          Key: '28143',
          LocalizedName: 'Dhaka',
          EnglishName: 'Dhaka',
          Country: { ID: 'BD', LocalizedName: 'Bangladesh', EnglishName: 'Bangladesh' },
          TimeZone: {
            Code: 'BDT', Name: 'Asia/Dhaka', GmtOffset: 6, IsDaylightSaving: false, NextOffsetChange: null,
          },
          GeoPosition: { Latitude: 23.7098, Longitude: 90.40711, Elevation: { Metric: { Value: 5, Unit: 'm', UnitType: 5 }, Imperial: { Value: 16, Unit: 'ft', UnitType: 0 } } },
          LocalObservationDateTime: '2023-08-05T08:37:00+06:00',
          EpochTime: 1691203020,
          WeatherText: 'Cloudy',
          WeatherIcon: 7,
          HasPrecipitation: false,
          PrecipitationType: null,
          IsDayTime: true,
          Temperature: { Metric: { Value: 28.5, Unit: 'C', UnitType: 17 }, Imperial: { Value: 83, Unit: 'F', UnitType: 18 } },
          MobileLink: 'http://www.accuweather.com/en/bd/dhaka/28143/current-weather/28143?lang=en-us',
          Link: 'http://www.accuweather.com/en/bd/dhaka/28143/current-weather/28143?lang=en-us',
        }],
        status: 'succeeded',
        error: null,
      },
    });
  });

  test('Should have the country send by mock store', () => {
    render(<Provider store={store}><App /></Provider>);
    const citiesListedElement = screen.getByText('We currently have 1 cities listed.');
    const amoutAppearOnScreen = screen.getByText('1');
    const continentImageIsPresent = screen.getByAltText('Asia');
    expect(citiesListedElement).toBeInTheDocument();
    expect(amoutAppearOnScreen).toBeInTheDocument();
    expect(continentImageIsPresent).toBeInTheDocument();
  });

  test('Should have all the continents on screen', () => {
    render(<Provider store={store}><App /></Provider>);
    const Africa = screen.getByAltText('Africa');
    const America = screen.getByAltText('America');
    const Asia = screen.getByAltText('Asia');
    const Europe = screen.getByAltText('Europe');
    const Oceania = screen.getByAltText('Oceania');
    expect(Africa).toBeInTheDocument();
    expect(America).toBeInTheDocument();
    expect(Asia).toBeInTheDocument();
    expect(Europe).toBeInTheDocument();
    expect(Oceania).toBeInTheDocument();
  });

  test('renders the App component correctly', () => {
    const { container } = render(<Provider store={store}><App /></Provider>);
    expect(container).toMatchSnapshot();
  });
});
