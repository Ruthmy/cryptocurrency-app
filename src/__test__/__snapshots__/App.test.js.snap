// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component correctly 1`] = `
<div>
  <nav
    class="d-flex-row"
  >
    <div>
      <a
        aria-current="page"
        class="active"
        href="#/"
      >
        <img
          alt="Back"
          class="nav__back"
          src="arrow.svg"
        />
        home
      </a>
    </div>
    <div
      class="nav__panel d-flex-row"
    >
      <button
        class="btn"
        type="button"
      >
        <img
          alt="Microphone"
          src="microphone.svg"
        />
      </button>
      <button
        class="btn"
        type="button"
      >
        <img
          alt="Settings"
          src="gear.svg"
        />
      </button>
    </div>
  </nav>
  <div
    class="home d-flex-row"
  >
    <img
      alt="Weather App Logo"
      class="header"
      src="logo.svg"
    />
    <div
      class="home__info"
    >
      <h1
        class="title"
      >
        WEATHER APP
      </h1>
      <p
        class="description"
      >
         We currently have 1 cities listed.
      </p>
    </div>
  </div>
  <div
    class="division"
  >
    <h2>
      CONTINENTS
    </h2>
  </div>
  <div
    class="continents d-flex-row"
  >
    <a
      class="continents__link d-flex-column"
      href="#/Africa"
    >
      <img
        alt="Africa"
        class="continents__image"
        src="Africa.svg"
      />
      <p
        class="continents__detail"
      >
        Africa
      </p>
      <p
        class="continents__cities"
      >
        0
      </p>
    </a>
    <a
      class="continents__link d-flex-column"
      href="#/America"
    >
      <img
        alt="America"
        class="continents__image"
        src="America.svg"
      />
      <p
        class="continents__detail"
      >
        America
      </p>
      <p
        class="continents__cities"
      >
        0
      </p>
    </a>
    <a
      class="continents__link d-flex-column"
      href="#/Asia"
    >
      <img
        alt="Asia"
        class="continents__image"
        src="Asia.svg"
      />
      <p
        class="continents__detail"
      >
        Asia
      </p>
      <p
        class="continents__cities"
      >
        1
      </p>
    </a>
    <a
      class="continents__link d-flex-column"
      href="#/Europe"
    >
      <img
        alt="Europe"
        class="continents__image"
        src="Europe.svg"
      />
      <p
        class="continents__detail"
      >
        Europe
      </p>
      <p
        class="continents__cities"
      >
        0
      </p>
    </a>
    <a
      class="continents__link d-flex-column"
      href="#/Australia"
    >
      <img
        alt="Oceania"
        class="continents__image"
        src="Oceania.svg"
      />
      <p
        class="continents__detail"
      >
        Oceania
      </p>
      <p
        class="continents__cities"
      >
        0
      </p>
    </a>
  </div>
</div>
`;
